### ISSUE TITLE
date is forgotten on reboot

### ISSUE THREAD

**Issue by networkimprov** [networkimprov/arch-packages#46]
date -s ... seems to work, but on reboot, the date is back to where it was.

**Comment by networkimprov** [comment]
@tmlind should there be a ref to RTC in .dts? I don't see one.

I got this msg well before systemd start until I built with `CONFIG_RTC_DRV_TWL4030=y`

`[    2.395050] hctosys: unable to open rtc device (rtc0)`

So the kernel wants to set system time from RTC early on.

**Comment by mranostay** [comment]
Yes there should be an entry for it (aka compatible = "ti,twl4030-rtc"). Only reason it would work as builtin is because it is a platform driver.

**Comment by networkimprov** [comment]
@tmlind any thoughts? If someone sends me an RTC entry for dts, I'll see if that fixes it...

**Comment by mranostay** [comment]
Actually the dts is including "twl4030.dtsi" so it should be enumerating.

**Comment by networkimprov** [comment]
The rtc driver loads later in boot when CONFIG...=m, but it's not soon enough to satisfy hctosys.

**Comment by mranostay** [comment]
Ok then like @tmlind mentioned we need to fix the systemd service

**Comment by networkimprov** [comment]
hctosys is a kernel feature! http://cateee.net/lkddb/web-lkddb/RTC_HCTOSYS.html

Believe me now? :-)

**Comment by mranostay** [comment]
Ok makes sense... @tmlind was probably talking about the systemd service that updates the RTC with ntp time, or in case of non-battery backed RTC.

**Comment by tmlind** [comment]
There used to be configurable /etc/init.d/hwclock.sh that also writes back the system time to RTC on shutdown and reboot if configured.. No idea what systemd might be doing nowadays with that

**Comment by networkimprov** [comment]
Also @tmlind suggested `alias { rtc0 = &rtc; };` for dts, but that didn't compile. What's the need for this?

**Comment by mranostay** [comment]
@networkimprov  Probably getting that message because   #include "twl4030.dtsi" is farther down the dts file.

**Comment by networkimprov** [comment]
Nope, checked that already.

**Comment by tmlind** [comment]
The alias is needed to specify the order of RTC controllers if more than one

**Comment by networkimprov** [comment]
@tmlind, could you add this to anvl-v4.7 branch when you get a chance?

**Comment by tmlind** [comment]
ah the twl dtsi file needed the twl_rtc: rtc alias. pushed now

**Comment by networkimprov** [comment]
The mmc0-3 aliases also added aren't used anywhere; which is fine, maybe they clarify what the original &mmcN terms mean :-)

**Comment by networkimprov** [comment]
@tmlind, can you make the twl dtsi fix a separate patch, since that's going upstream?

**Comment by networkimprov** [comment]
@tmlind, disregard twl dtsi patch request; just make sure it goes upstream :-)
